src/gui/src/timingWidget.cpp

```diff
@@ -0,0 +1,394 @@
+      this,
+      SLOT(selectedRowChanged(const QItemSelection&, const QItemSelection&)));
+
+  connect(
+      hold_timing_table_view_->selectionModel(),
+      SIGNAL(selectionChanged(const QItemSelection&, const QItemSelection&)),
+      this,
+      SLOT(selectedRowChanged(const QItemSelection&, const QItemSelection&)));
+
+  connect(
+      path_details_table_view_->selectionModel(),
+      SIGNAL(selectionChanged(const QItemSelection&, const QItemSelection&)),
+      this,
+      SLOT(selectedDetailRowChanged(const QItemSelection&,
+                                    const QItemSelection&)));
+
+  connect(render_enable_button_, SIGNAL(toggled(bool)), this, SLOT(toggleRenderer()));
+
+  connect(setup_timing_paths_model_, SIGNAL(modelReset()), this, SLOT(modelWasReset()));
+  connect(hold_timing_paths_model_, SIGNAL(modelReset()), this, SLOT(modelWasReset()));
+
+  path_index_spin_box_->setRange(0, 0);
+}
+
+TimingWidget::~TimingWidget()
+{
+}
+
+void TimingWidget::readSettings(QSettings* settings)
+{
+  settings->beginGroup(objectName());
+
+  render_enable_button_->setChecked(settings->value("draw_enabled", render_enable_button_->isChecked()).toBool());
+  path_count_spin_box_->setValue(settings->value("path_count", path_count_spin_box_->value()).toInt());
+
+  settings->endGroup();
+}
+
+void TimingWidget::writeSettings(QSettings* settings)
+{
+  settings->beginGroup(objectName());
+
+  settings->setValue("draw_enabled", render_enable_button_->isChecked());
+  settings->setValue("path_count", path_count_spin_box_->value());
+
+  settings->endGroup();
+}
+
+void TimingWidget::keyPressEvent(QKeyEvent* key_event)
+{
+  if (key_event->matches(QKeySequence::Copy)) {
+    copy();
+    key_event->accept();
+  }
+}
+
+void TimingWidget::showPathDetails(const QModelIndex& index)
+{
+  if (!index.isValid())
+    return;
+
+  if (index.model() == setup_timing_paths_model_) {
+    focus_view_ = setup_timing_table_view_;
+    hold_timing_table_view_->clearSelection();
+  } else {
+    focus_view_ = hold_timing_table_view_;
+    setup_timing_table_view_->clearSelection();
+  }
+
+  TimingPathsModel* focus_model = static_cast<TimingPathsModel*>(focus_view_->model());
+
+  auto path = focus_model->getPathAt(index.row());
+
+  path_details_model_->populateModel(path);
+  path_details_table_view_->resizeColumnsToContents();
+  path_details_table_view_->horizontalHeader()->setSectionResizeMode(0, QHeaderView::Stretch);
+
+  path_renderer_->highlight(path);
+  emit highlightTimingPath(path);
+
+  path_index_spin_box_->setRange(0, focus_model->rowCount()-1);
+  path_index_spin_box_->setValue(index.row());
+}
+
+void TimingWidget::highlightPathStage(const QModelIndex& index)
+{
+  if (!index.isValid()) {
+    return;
+  }
+  path_renderer_->highlightNode(index.row());
+  emit highlightTimingPath(path_renderer_->getPathToRender());
+}
+
+void TimingWidget::timingPathsViewCustomSort(int col_index)
```

Comment:
I saw this but when I tried to click the sort icons nothing happens.

Reply:
I see what you mean, I didn't modify any of that so I would be interested to know if it was working before as well?
Currently it's limited to just a single column for sorting, but it might be nice to expand that to more.

Reply:
You could try the head of master and see.  I've never tried sorting before but saw it in the changes so I thought I should try it.  It should either work or be disabled but if you want to defer it that's ok as it isn't changed by this PR.